TELCODOCS-1092: GPU sharing methods (#67322)
openshift/openshift-docs: modules/nvidia-gpu-time-slicing.adoc

// Module included in the following assemblies:
//
// * architecture/nvidia-gpu-architecture-overview.adoc

:_content-type: CONCEPT
[id="nvidia-gpu-time-slicing_{context}"]
= Time-slicing

> GPU time-slicing enables workloads scheduled on overloaded GPUs to be interleaved when you are running multiple CUDA applications.

Reviewer: ```suggestion
GPU time-slicing interleaves workloads scheduled on overloaded GPUs when you are running multiple CUDA applications.
```
Rewording for active voice

> The mechanism for enabling time-slicing of GPUs in Kubernetes allows you to define a set of replicas for a GPU, each of which can be independently distributed to a pod to run workloads on. Unlike multi-instance GPU (MIG), there is no memory or fault isolation between replicas, but for some workloads this is better than not sharing at all. Internally, GPU time-slicing is used to multiplex workloads from replicas of the same underlying GPU.

Reviewer: ```suggestion
You can enable time-slicing of GPUs on Kubernetes by defining a set of replicas for a GPU, each of which can be independently distributed to a pod to run workloads on. Unlike multi-instance GPU (MIG), there is no memory or fault isolation between replicas, but for some workloads this is better than not sharing at all. Internally, GPU time-slicing is used to multiplex workloads from replicas of the same underlying GPU.
```

Per IBM Style guide:
"Avoid stating that inanimate objects grant abilities to people, as in “the product allows you to…”. Whenever possible, use a direct, user-focused alternative such as “you can use the product to…” or “with this product, users can…”."

// Module included in the following assemblies:
//
// * architecture/nvidia-gpu-architecture-overview.adoc

:_content-type: CONCEPT
[id="nvidia-gpu-time-slicing_{context}"]
= Time-slicing

When dealing with multiple CUDA applications, each of which may not fully utilize the GPU’s resources,
you can use a simple oversubscription strategy to leverage the GPU’s time-slicing scheduler.
This technique, sometimes called temporal GPU sharing, does carry a cost for context-switching between
the different CUDA applications, but some underutilized applications can still benefit from this strategy.

The tradeoffs with time-slicing are increased latency, jitter, and potential out-of-memory (OOM) conditions
when many different applications are time-slicing on the GPU.

GPU time-slicing enables workloads scheduled on overloaded GPUs to be interleaved.

> This mechanism for enabling time-slicing of GPUs in Kubernetes enables a system administrator to define a set of replicas for a GPU, each of which can be independently distributed to a pod to run workloads on. Unlike multi-instance GPU (MIG), there is no memory or fault isolation between replicas, but for some workloads this is better than not sharing at all. Internally, GPU time-slicing is used to multiplex workloads from replicas of the same underlying GPU.

Reviewer 2: Similar question about attribution for the text for these 3 paragraphs that seem to be from this link: https://docs.nvidia.com/datacenter/cloud-native/gpu-operator/latest/gpu-sharing.html

Author: Removed first 3 paragraphs and replaced with text from a topic similar topic that I removed from the NVIDIA GPU features section because of rendundancy. 

"GPU time-slicing enables workloads scheduled on overloaded GPUs to be interleaved when you are running multiple CUDA applications.

The mechanism for enabling time-slicing of GPUs in Kubernetes allows you to define a set of replicas for a GPU, each of which can be independently distributed to a pod to run workloads on. Unlike multi-instance GPU (MIG), there is no memory or fault isolation between replicas, but for some workloads this is better than not sharing at all. Internally, GPU time-slicing is used to multiplex workloads from replicas of the same underlying GPU.

You can apply a cluster-wide default configuration for time-slicing. You can also apply node-specific configurations. For example, you can apply a time-slicing configuration only to nodes with Tesla T4 GPUs and not modify nodes with other GPU models.

You can combine these two approaches by applying a cluster-wide default configuration and then labeling nodes to give those nodes a node-specific configuration.
